api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -380,58 +381,74 @@ public Builder identity(String sourceName) {
       return identity(sourceName, sourceName);
     }
 
-    public Builder year(String sourceName, String targetName) {
+    public Builder year(String sourceName, String targetName, ZoneOffset zoneOffset) {
```

Comment:
I would rather not expose Java's time library in Iceberg's API. Could we use a string instead?